### ISSUE TITLE
LabelDirective: Cannot read property `focus` of undefined

### ISSUE THREAD

**Issue by geo242** [telerik/kendo-angular#3239]
**Describe the bug**
The LabelDirective produces the following error if the field is dynamically generated or the [for] input is not set. This only happens when the kendo-label does NOT surround the input element.

This seems to be a regression in v3.0.1

`
ERROR TypeError: Cannot read property 'focus' of undefined
    at LabelDirective.getFocusableComponent (index.js:183)
    at LabelDirective.get labelFor [as labelFor] (index.js:138)
    at LabelDirective_HostBindings (index.js:188)
    at processHostBindingOpCodes (core.js:9186)
    at refreshView (core.js:9464)
    at refreshComponent (core.js:10584)
    at refreshChildComponents (core.js:9215)
    at refreshView (core.js:9468)
    at refreshEmbeddedViews (core.js:10538)
    at refreshView (core.js:9442)
`

    getFocusableComponent() {
        const target = this.for;
        return target.focus !== undefined ? target : null;
    }

**To Reproduce**
https://stackblitz.com/edit/bk6e8y--run?file=app/app.component.ts

remove the [for] input on line 16 of app.component.ts

**Expected behavior**
Should gracefully handle missing element and not throw an error.

**Browser**
Any browser

**Problem Code**

```
getFocusableComponent() {
  const target = this.for; // Error occurs when target is undefined
  return target.focus !== undefined ? target : null;
}
```

*Fix*
```
getFocusableComponent() {
  const target = this.for || {};
  return target.focus !== undefined ? target : null;
}
```

**Comment by jim-typhongroup** [comment]
For the record, this bug is causing issues with our production build as well!   We manually added the fix above to the index.js file get around it for now.

**Comment by geo242** [comment]
@jim-typhongroup We downgraded to v3.0.0 until this is fixed. That might be a more maintainable solution.

**Comment by Raisolution** [comment]
Hi guys,

The issue has been addressed in `v3.0.2`.
